packages/components/src/types/page.ts

```diff
@@ -10,6 +10,39 @@ import {
 import { imageSchemaObject } from "~/schemas/internal"
 import { REF_HREF_PATTERN } from "~/utils/validation"
 
+// NOTE: a tag value is simply a uuid that maps to a given label;
+// essentially, it is just a pointer
+const UuidSchema = Type.String({ format: "uuid" })
+export const TagValueSchema = UuidSchema
+// NOTE: single value for now but we might extend this in the future with additional metadata,
+// so we will leave it as is
+const DropdownItemSchema = Type.Object({ label: Type.String(), id: UuidSchema })
+const TagOptionSchema = DropdownItemSchema
+const TagCategorySchema = Type.Composite([
+  Type.Object({
+    options: Type.Array(TagOptionSchema),
+  }),
+  DropdownItemSchema,
+])
+// NOTE: can be optional because the categories might not exist
+const TagCategoriesSchema = Type.Object({
+  tagCategories: Type.Optional(Type.Array(TagCategorySchema)),
+})
+const TaggedSchema = Type.Optional(
+  // NOTE: This stores the `uuid` of the tag category
+  // to the array of the `uuid` of the values;
+  // we cannot just store a plain array of `uuid`
+  // because we need to render each category as a dropdown
+  Type.Array(
+    Type.Object({ id: TagValueSchema, values: Type.Array(TagValueSchema) }),
```

Comment:
nit: can we rename these to be something like TagCategoryIdSchema and TagOptionIdSchema to distinguish their intent?

Or actually can make use of the `title` and `description` keys to elaborate more on each property of the schema, so someone reading the schema on [schema.isomer.gov.sg](https://schema.isomer.gov.sg) can make immediate sense of it